src/anatomy.cpp

```diff
@@ -211,3 +214,56 @@ bodypart_id anatomy::select_body_part( int min_hit, int max_hit, bool can_attack
     add_msg_debug( debugmode::DF_ANATOMY_BP, "selected part: %s", ret->id().obj().name );
     return *ret;
 }
+
+
+bodypart_id anatomy::select_blocking_part( bool arm, bool leg, bool nonstandard ) const
+{
+    weighted_float_list<bodypart_id> block_scores;
+    for( const bodypart_id &bp : cached_bps ) {
+        float block_score = bp->get_limb_score( limb_score_block );
```

Comment:
`body_part_type::get_limb_score` just returns the bare score. What you want is `bodypart::get_limb_score` or (recommended) `Character::get_limb_score` and pass in the body part type.

The reason is because the id points to a `body_part_type`, which is a constant (never changing) base for body parts to inherit from. The modified values live in `bodypart`, which is a dynamic object that modifies the values based on the parent character.

In this case, you'd need to get the `bodypart` from the character and call get_limb_score from there. I'm sorry it's not intuitive, the encumbrance and wound modifiers live in `bodypart` and not `body_part_type`.

Reply:
Oops, looks like you've already figured as much (sorry!). The way to fix this would be to somehow pass the character into the function or make the anatomy aware of the host. It might also be easier to do the processing of this function elsewhere.

Once I get some time, I'll give it a closer look.